### PR TITLE
Add documentation for note filters and task trace guides

### PR DESCRIPTION
**Pull request by YuuichiNakamura**
## Summary
This PR adds the documentation of note filters and task trace function which is introduced in #1936 and https://github.com/apache/incubator-nuttx-apps/pull/421

### CODE REVIEW

**Comment by YuuichiNakamura** [comment]
@v01d @xiaoxiang781216 I have added the document for note filters.
This is an introduction of only filter device. The application command introduction will be added in https://github.com/apache/incubator-nuttx-apps/pull/421 as README.md

**Comment by v01d** [comment]
The docs look fine, albeit there should also be some explanation inside the first section of *what* the note driver does. Keep in mind that not everyone (as me) is aware of what it does and how it works.
I was looking forward to have the docs you linked here: https://github.com/YuuichiNakamura/nuttx-task-tracer-doc
with all those nice figures and explanations.

**Comment by YuuichiNakamura** [comment]
> The docs look fine, albeit there should also be some explanation inside the first section of _what_ the note driver does. Keep in mind that not everyone (as me) is aware of what it does and how it works.
> I was looking forward to have the docs you linked here: https://github.com/YuuichiNakamura/nuttx-task-tracer-doc
> with all those nice figures and explanations.

Ok. I'll add an explanation of note driver on the top of note.rst.
Regarding the another part of my document, how about adding them to Documentation/guides ? It seems a preferable place for the introduction. 

And, should I leave a instruction of "trace" command line options in apps/system/trace/README.md ? I couldn't find a preferable place to explain the built-in command line options in Documentation/. Or should I move it into Documentation/guides with other instructions (Installation, Quick start guide, ... etc)

**Comment by v01d** [comment]
> > The docs look fine, albeit there should also be some explanation inside the first section of _what_ the note driver does. Keep in mind that not everyone (as me) is aware of what it does and how it works.
> > I was looking forward to have the docs you linked here: https://github.com/YuuichiNakamura/nuttx-task-tracer-doc
> > with all those nice figures and explanations.
> 
> Ok. I'll add an explanation of note driver on the top of note.rst.
> Regarding the another part of my document, how about adding them to Documentation/guides ? It seems a preferable place for the introduction.

I'd always prefer to complete the reference sections before adding stuff to the guides. But if you prefer to have it there, go ahead.
The main image could be added to the doc page of this PR in any case.

> 
> And, should I leave a instruction of "trace" command line options in apps/system/trace/README.md ? I couldn't find a preferable place to explain the built-in command line options in Documentation/. Or should I move it into Documentation/guides with other instructions (Installation, Quick start guide, ... etc)

Builtin commands are described here: https://nuttx.apache.org/docs/latest/components/nsh/commands.html
This section is fairly large and eventually should be split for readability.
Remember that you can use the search to find where things are documented.

**Comment by YuuichiNakamura** [comment]
@v01d I have updated the commit to incorporate all contents in  https://github.com/YuuichiNakamura/nuttx-task-tracer-doc
Currently apps/system/trace/README.md has the same contents to tasktraceuser.rst.
To reduce the maintenance cost, I'll fix the former to the reference to rst version later.

**Review by v01d — APPROVED** [review]

File: Documentation/guides/tasktraceuser.rst [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    Higher value can hold more note records, but consumes more kernel memory.
+
+- ``CONFIG_SCHED_INSTRUMENTATION_NOTERAM_DEFAULT_NOOVERWRITE``
+
+  - If enabled, stop overwriting old notes in the circular buffer when the buffer is full by default.
+    This is useful to keep instrumentation data of the beginning of a system boot.
+
+After the configuration, rebuild the NuttX kernel and application.
+
+If the trace function is enabled, "``trace``" :doc:`../components/nsh/builtin` will be available.
```
Comment (xiaoxiang781216) [inline Documentation/guides/tasktraceuser.rst:57]: not implement as nsh builtin now.

Comment (YuuichiNakamura) [reply Documentation/guides/tasktraceuser.rst:57]: `NSH “Built-In” Applications` is different to NSH commands.

> In addition to these commands that are a part of NSH, external programs can also be executed as NSH commands. These external programs are called "Built-In" Applications for historic reasons.

Based on this description in components/nsh/builtin.rst, "trace" is a built-in application.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**